checker-qual/src/main/java/org/checkerframework/checker/initialization/qual/PolyInitialized.java

```diff
@@ -0,0 +1,29 @@
+package org.checkerframework.checker.initialization.qual;
+
+import org.checkerframework.framework.qual.PolymorphicQualifier;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A polymorphic qualifier for the freedom-before-commitment initialization tracking type-system.
+ *
+ * <p>Any method written using {@link PolyInitialized} takes a class argument and is instantiated
```

Comment:
This says that the method takes a class argument, which isn't what will happen.
And that class argument isn't the three initialization annotations, but instead the type frame of that particular use.

In a comment that is now hard to see, you said "It looks like PolyRegex, PolyInterned, PolyNull and PolySignature all has the problem".
In the nullness system, none of the annotations (neither `@Nullable` nor `@NonNull`) take an annotation attribute/argument. So the problem doesn't exist there. In contrast, here and e.g. in `@PolyRegex`, there are annotations with further attributes, e.g. one could write `@Regex(2)` or `@UnderInitialization(Bound.class)`.
So saying that there is a copy of the method for each qualifier is incorrect - conceptually, there is a copy of the method for each possible instantiation of each qualifier.

Does this make more sense now?

Reply:
Yes, thanks for explanation